core/src/main/scala/za/co/absa/commons/CaptureGroupReplacer.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.commons
+
+import scala.util.matching.Regex
+
+class CaptureGroupReplacer(replacement: String) {
+
+  def replace(str: String, regexes: Seq[Regex]): String = {
+    if (regexes.isEmpty)
+      str
+    else {
+      val nextStr = regexes.head
+        .findFirstMatchIn(str)
+        .map(regMatch => replaceMatchedGroups(regMatch, str))
+        .getOrElse(str)
+
+      replace(nextStr, regexes.tail)
+    }
+  }
```

Comment:
Decomposing a collection with pattern-matching is more idiomatic and readable than `.isEmpty`, `.head` and `.tail` combination.
Also `@tailrec` makes a good fit here.
```suggestion
  @tailrec
  final def replace(str: String, regexes: Seq[Regex]): String = regexes match {
    case Nil => str
    case regex :: nextRegexes =>
      val nextStr = regex
        .findFirstMatchIn(str)
        .map(regMatch => replaceMatchedGroups(regMatch, str))
        .getOrElse(str)
      replace(nextStr, nextRegexes)
  }
```